src/Service/AlgoliaIndexer.php

```diff
@@ -162,8 +162,10 @@ public function exportAttributesFromObject($item)
             'objectCreated' => $item->dbObject('Created')->getTimestamp()
         ];
 
-        if ($item->hasMethod('AbsoluteLink')) {
+        if ($item->hasMethod('AbsoluteLink') && !empty($item->AbsoluteLink())) {
             $toIndex['objectLink'] = str_replace(['?stage=Stage', '?stage=Live'], '', $item->AbsoluteLink());
```

Comment:
This is going to call `AbsoluteLink` twice for every record indexing. While that can be a quick method it may also do quite a few look ups (subsite etc). Can we save the value to a value after the hasMethod  `$link = $item->AbsoluteLink();` and use that `$link` on the line below. Likewise for link.

Reply:
Updated now